# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../guile/gcc-4.0.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Taken from Fedora Core.

--- guile-1.6.7/libguile/guile.c.ltdl	2005-03-02 16:31:46.000000000 +0100
+++ guile-1.6.7/libguile/guile.c	2005-03-02 16:32:07.000000000 +0100
@@ -53,6 +53,7 @@
 #endif
 
 #ifdef DYNAMIC_LINKING
+#include <ltdl.h>
 #include "guile-ltdl.h"
 #endif
 
@@ -88,7 +89,7 @@
 {
 #ifdef DYNAMIC_LINKING
   /* libtool automagically inserts this variable into your executable... */
-  extern const scm_lt_dlsymlist lt_preloaded_symbols[];
+  extern const lt_dlsymlist lt_preloaded_symbols[];
   scm_lt_dlpreload_default (lt_preloaded_symbols);
 #endif
   scm_boot_guile (argc, argv, inner_main, 0);
